app/controllers/client_data_controller.rb

```diff
@@ -94,12 +94,20 @@ def auth_errors(exception)
     path = polymorphic_path(model_class, action: :new)
     # prevent redirect loop
     if path == request.path
-      render "authorization_error", status: 403, locals: { msg: exception.message }
+      render_auth_errors(exception)
     else
       redirect_to path, alert: exception.message
     end
   end
 
+  def render_auth_errors(exception)
+    if exception.message == "Client is disabled"
+      render_disabled_client_message(exception.message)
```

Comment:
If there is an exception in the `client_data_controller` that is *not* a disabled-client error, we want to render the 403 authz error page. That is the existing behavior.

I'm not sure exactly what you think is being duplicated? I've basically just refactored all the rendering code in the various controllers into its own methods so that rubocop likes me.